update font optimization page (#42266)
vercel/next.js: docs/basic-features/font-optimization.md

Import the font you would like to use from `@next/font/google` as a function. We recommend using [**variable fonts**](https://fonts.google.com/variablefonts) for the best performance and flexibility.

> To use the font in all your pages, add it to `_app.js` or `_app.tsx` under `/pages` as shown below:

Reviewer: ```suggestion
To use the font in all your pages, add it to [`_app.js` file](https://nextjs.org/docs/advanced-features/custom-app) under `/pages` as shown below:
```

// If loading a variable font, you don't need to specify the font weight
const inter = Inter()

export default function MyApp({ Component, pageProps }) {
  return (
    <main className={inter.className}>
      <Component {...pageProps} />
    </main>
  )
}
```

If you can't use a variable font, you will **need to specify a weight**:

```js:pages/_app.js
import { Roboto } from '@next/font/google';

const roboto = Roboto({
  weight: '400',
})

export default function MyApp({ Component, pageProps }) {
  return (
    <main className={roboto.className}>
      <Component {...pageProps} />
    </main>
  )
}
```

#### Apply the font in `<head>`

You can also use the font without a wrapper and `className` by injecting it inside the `<head>` as follows:

```js:pages/_app.js
import { Inter } from '@next/font/google';

const inter = Inter();

export default function MyApp({ Component, pageProps }) {
  return (
    <>
      <style jsx global>{`
        html {
          font-family: ${inter.style.fontFamily};
        }
      `}</style>
      <Component {...pageProps} />
    </>
  )
}
```

#### Single page usage

To use the font on a single page, add it to the specific page as shown below:

> ```js:pages/index.js

Reviewer 2: Should the example use the font in a component that the page renders, to make it clear that you don't have to call the font function in the actual page for it to work? What do you think?
```js:pages/index.js
import Header from "../components/header.js"

export default function Home() {
  return (
    <>
        <Header>Test</Header>
        <p>Hello World</p>
    </>
  );
}
```

```js:components/header.js
import { inter } from "../styles/fonts.js"

export default function Header({ children }) {
  return <h1 className={inter.className}>{children}</h1>;
}
```

    <small>Learn how to optimize images with the Image component.</small>
  </a>
</div>

> ````

Reviewer: ```suggestion
```
```

> ````

Reviewer: ```suggestion
```
